src/elements/List/ListItem.js

```diff
@@ -50,11 +93,34 @@ ListItem.propTypes = {
   /** Additional classes. */
   className: PropTypes.string,
 
+  /** Shorthand for primary content. */
+  content: customPropTypes.contentShorthand,
+
+  /** Shorthand for ListDescription. */
+  description: customPropTypes.itemShorthand,
+
   /** A list item can disabled. */
   disabled: PropTypes.bool,
 
+  /** Shorthand for ListHeader. */
+  header: customPropTypes.itemShorthand,
+
+  /** Shorthand for ListIcon. */
+  icon: customPropTypes.every([
```

Comment:
What do you say we add `iconShorthand` and `imageShorthand` that can handle disallowing the each other?  This is also needed in the header, perhaps elsewhere.